### ISSUE TITLE
No ability to override new belongs_to navigation menu functionality

### ISSUE THREAD

**Issue by onyxraven** [activeadmin/activeadmin#2637]
Currently, it looks like there is no way to avoid the behavior of the navigation_menu being overridden to the belongs_to parent, as well as the menu item being automatically added and highlighted in the menu

That is, effectively when you have a resource with belongs_to, you get

``` ruby
ActiveAdmin.register Object do
  belongs_to :parent_object
  navigation_menu :parent_object
  menu "Objects"
end
```

This isn't evident until you visit that belongs_to controller, it adds all the resource that share the same belongs_to to the menu.

The old behavior was that if you had a belongs_to object that had no menu DSL statement, it would never be added to the menu (effectively menu false), and the menu would stay the default/global menu.

I've got a icky patch I've worked around to get 'back' to the old behavior.

``` ruby
module ActiveAdmin
  class ResourceDSL < DSL

    def run_registration_block(&block)
      #stuff before normal dsl that can be overridden
      navigation_menu :default

      super

      return unless block_given? and defined? config and config

      if config.belongs_to? and config.menu_item_options[:parent].blank?
        navigation_menu :default
        menu false
      end
    end

  end
end
```

So far, this seems to preserve the old behavior, but still enable the 'highlighting' of the menu item like previous.  I know this isn't a great solution but I hope it'll help get a config option started?

**Comment by jackxu** [comment]
Yes, hope the guys who can fix it...

**Comment by johnnagro** [comment]
:boat:

**Comment by kaelumania** [comment]
come on :+1:

**Comment by reneweteling** [comment]
👍  still an issue

**Comment by tombeynon** [comment]
So far the closest I've got is to just use the following config against the resource. This leaves the default menu in place at all times, and doesn't throw an error if the parent ID param is missing. 
```
navigation_menu :default
menu false
```

**Comment by Fivell** [comment]
@varyonic , it is our documentation issue, I'll do fix next days

**Comment by Fivell** [comment]
@varyonic , or we should leave docs and fix AA to have same behaviour.

> In some cases (like Projects), there are many sub resources and you would actually like the global navigation to switch when the user navigates "into" a project. To accomplish this, Active Admin stores the belongs_to resources in a separate menu which you can use if you so wish. To use:

```
ActiveAdmin.register Ticket do
  belongs_to :project
  navigation_menu :project
end
```
```
ActiveAdmin.register Milestone do

  belongs_to :project
  navigation_menu :project
end
```

> Now, when you navigate to the tickets section, the global navigation will only display "Tickets" and "Milestones". When you navigate back to a non-belongs_to resource, it will switch back to the default menu.


instead of this 

```
  navigation_menu :project
```

is default behaviour

**Comment by Fivell** [comment]
closed via #4842
